src/main/java/com/splunk/ibm/mq/metricscollector/ListenerMetricsCollector.java

```diff
@@ -76,43 +77,40 @@ public void publishMetrics() {
             "sending PCF agent request to query metrics for generic listener {}",
             listenerGenericName);
         long startTime = System.currentTimeMillis();
-        PCFMessage[] response = context.send(request);
+        List<PCFMessage> response = context.send(request);
         long endTime = System.currentTimeMillis() - startTime;
         logger.debug(
             "PCF agent listener metrics query response for generic listener {} received in {} milliseconds",
             listenerGenericName,
             endTime);
-        if (response == null || response.length <= 0) {
-          logger.debug(
-              "Unexpected Error while PCFMessage.send(), response is either null or empty");
+        if (response.isEmpty()) {
+          logger.debug("Unexpected error while PCFMessage.send(), response is empty");
           return;
         }
-        for (PCFMessage pcfMessage : response) {
-          String listenerName =
-              pcfMessage.getStringParameterValue(CMQCFC.MQCACH_LISTENER_NAME).trim();
-          Set<ExcludeFilters> excludeFilters = context.getListenerExcludeFilters();
-          if (!ExcludeFilters.isExcluded(
-              listenerName, excludeFilters)) { // check for exclude filters
-            logger.debug("Pulling out metrics for listener name {}", listenerName);
-            List<Metric> responseMetrics = Lists.newArrayList();
-            context.forEachMetric(
-                (metricKey, wmqOverride) -> {
-                  int metricVal = pcfMessage.getIntParameterValue(wmqOverride.getConstantValue());
-                  Metric metric =
-                      metricCreator.createMetric(
-                          metricKey, metricVal, wmqOverride, listenerName, metricKey);
-                  responseMetrics.add(metric);
-                });
-            context.transformAndPrintMetrics(responseMetrics);
-          } else {
-            logger.debug("Listener name {} is excluded.", listenerName);
-          }
+
+        List<PCFMessage> messages =
+            MessageFilter.ofKind("listener")
+                .excluding(context.getListenerExcludeFilters())
+                .withResourceExtractor(MessageBuddy::listenerName)
+                .filter(response);
+
+        for (PCFMessage message : messages) {
+          String listenerName = MessageBuddy.listenerName(message);
+          logger.debug("Pulling out metrics for listener name {}", listenerName);
+          List<Metric> responseMetrics = Lists.newArrayList();
+          context.forEachMetric(
```

Comment:
if we are refactoring the previous two places, we might as well do this one too.